dspace-api/src/main/java/org/dspace/external/provider/ExternalDataProvider.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.external.provider;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.dspace.external.model.ExternalDataObject;
+
+/**
+ * This interface should be implemented by all providers that will deal with external data
+ */
+public interface ExternalDataProvider {
+
+    /**
+     * This method will return the SourceIdentifier for the ExternalDataProvider that implements the interface
+     * @return  The source identifier as a String
+     */
+    public String getSourceIdentifier();
+
+    /**
+     * This method will take a String id as a parameter and it'll call the ExternalDataProvider's endpoint or data
+     * source to retrieve and build the ExternalDataObject
+     * @param id    The id on which will be searched
+     * @return      An Optional object of ExternalDataObject. This is to indicate that this object may be null.
+     *              This ExternalDataObject will return all the data returned by the ExternalDataProvider
+     */
+    Optional<ExternalDataObject> getExternalDataObject(String id);
+
+    /**
+     * This method will query the ExternalDataProvider's endpoint or data source to retrieve and build a list of
+     * ExternalDataObjects through a search with the given parameters
+     * @param query The query for the search
+     * @param start The start of the search
+     * @param limit The max amount of records to be returned by the search
+     * @return      A list of ExternalDataObjects that were retrieved and built by this search
+     */
+    List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit);
```

Comment:
A generic 500 error deals with any potential issue (doesn’t specify any details)
A 404 assumes the specific resource doesn’t exist, but is not applicable if e.g. the connection to ORCID failed
To avoid over-complicating potential reasons why an external source is causing an error, I'd prefer to keep a generic error at least for the first implementation.
If there's a decent set of integrations, it can be relevant to verify if there are some common exceptions which can be treated separately